Nodejs/Product/TestAdapter/TestDiscovererWorker.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.NodejsTools.SourceMapping;
+using Microsoft.NodejsTools.TestAdapter.TestFrameworks;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Adapter;
+using Microsoft.VisualStudio.TestPlatform.ObjectModel.Logging;
+using Microsoft.VisualStudioTools;
+
+namespace Microsoft.NodejsTools.TestAdapter
+{
+    public sealed class TestDiscovererWorker
+    {
+        private readonly string testSource;
+        private readonly string workingDir;
+        private readonly Uri testExecutorUri;
+        private readonly string nodeExePath;
+
+        public TestDiscovererWorker(string testSource, Uri testExecutorUri, string nodeExePath)
+        {
+            this.testSource = testSource;
+            this.workingDir = Path.GetDirectoryName(this.testSource);
+            this.testExecutorUri = testExecutorUri;
+            this.nodeExePath = nodeExePath;
+        }
+
+        public void DiscoverTests(string testFolderPath, TestFramework testFx, IMessageLogger logger, ITestCaseDiscoverySink discoverySink)
+        {
+            if (!File.Exists(this.nodeExePath))
+            {
+                logger.SendMessage(TestMessageLevel.Error, "Node.exe was not found. Please install Node.js before running tests.");
+                return;
+            }
+
+            var fileList = Directory.GetFiles(testFolderPath, "*.js", SearchOption.AllDirectories);
+            this.DiscoverTests(fileList, testFx, logger, discoverySink);
+        }
+
+        public void DiscoverTests(IEnumerable<string> fileList, TestFramework testFx, IMessageLogger logger, ITestCaseDiscoverySink discoverySink)
+        {
+            var files = string.Join(";", fileList);
+            logger.SendMessage(TestMessageLevel.Informational, $"Processing: {files}");
+
+            var discoveredTestCases = testFx.FindTests(fileList, this.nodeExePath, logger, projectRoot: this.workingDir);
```

Comment:
nodeExePath is only validated by the callers. Can it be removed from there and validated on this method?

Reply:
yes